metricflow/sql/optimizer/rewriting_sub_query_reducer.py

```diff
@@ -276,6 +280,32 @@ def _current_node_can_be_reduced(self, node: SqlSelectStatementNode) -> bool:
         ) > 0 and not SqlRewritingSubQueryReducerVisitor._select_columns_are_column_references(node.select_columns):
             return False
 
+        # If the group by is referencing a window function in the parent node, it can't be reduced.
+        # Window functions can't be included in group by.
+        parent_column_aliases_with_window_functions = {
+            select_column.column_alias
+            for select_column in SqlRewritingSubQueryReducerVisitor._select_columns_with_window_functions(
+                parent_select_node.select_columns
+            )
+        }
+        if len(node.group_bys) > 0 and [
+            group_by
+            for group_by in node.group_bys
+            if (
+                group_by.column_alias in parent_column_aliases_with_window_functions
+                or (
+                    group_by.expr.as_column_reference_expression
+                    and group_by.expr.as_column_reference_expression.col_ref.column_name
+                    in parent_column_aliases_with_window_functions
+                )
+                or (
+                    group_by.expr.as_string_expression
+                    and group_by.expr.as_string_expression.sql_expr in parent_column_aliases_with_window_functions
+                )
```

Comment:
Oof, why are there so many ways to reference a group by? Oh well. I think I'll blame BigQuery and Trino on general principles.